src/userAgent.ts

```diff
@@ -0,0 +1,22 @@
+import { AxiosHeaders } from "axios";
+import { ExtendedAxiosRequestConfig } from "./NetworkProviderConfig";
+
+export function setUserAgent(userAgentPrefix: string, config: ExtendedAxiosRequestConfig | undefined) {
```

Comment:
`extendUserAgent` or something similar (we are not clearing what has been previously set)?

Reply:
We should add a small unit test.

Reply:
done, tests added